src/utils/setContainerStyles.ts

```diff
@@ -0,0 +1,19 @@
+import breakpoints from "../theme/foundations/breakpoints";
+
+/**
+ * Generates container query styles based on the provided breakpoint and styles.
+ *
+ * @param {string} props.breakpoint - The breakpoint at which the container query should apply. It can be "base" or a key from the breakpoints object.
+ * @param {object} props.styles - The styles to apply within the container query.
+ * @returns {object} An object containing the container query styles.
+ */
+export const setContainerStyles = (props: {
```

Comment:
This is great. At first, I was a bit hesitant about this. It seemed like a bit more added complexity for an already complicated CSS way of adding styles. The initial approach of adding the `@containers` rules in the style files made more to me as I was reading and understanding container queries. This adds an easier way to write them, but hope it's not more complicated when learning container queries.

Reply:
I think there is a ticket for writing up a Storybook page for a guide to adding container queries to components. We can clarify any complicated approaches there.